examples/EmailerAgent/emailer/agent.py

```diff
@@ -0,0 +1,419 @@
+
+        self.vip.config.subscribe(self.configure_main,
+                                  actions=["NEW", "UPDATE"], pattern="config")
+
+        # Keep track of keys that have been added to send with.
+        self.tosend = {}
+
+        def onstart(sender, **kwargs):
+            self.vip.pubsub.subscribe('pubsub', topics.PLATFORM_SEND_EMAIL,
+                                      self.on_email_message)
+
+            self.vip.pubsub.subscribe('pubsub', topics.ALERTS_BASE,
+                                      self.on_alert_message)
+            self.vip.pubsub.subscribe('pubsub',
+                                      prefix=topics.ALERTS.format(agent_class='',
+                                                                  agent_uuid=''),
+                                      callback=self.on_alert_message)
+
+        self.core.onstart.connect(onstart, self)
+
+        self.sent_alert_emails = defaultdict(int)
+
+#Add this function to test the emailer by publishing to the topic and passing messages and headers 
+#    @Core.periodic(60)
+#    def accessService(self):
+#        headers = {
+#            "something": 'something'
+#            #"to-addresses": ['alpha.beta@foo.com', 'bob-and-joe@bar.com']
+#        }
+
+#        message = {
+#            "subject": "Something is not right",
+#            "message": "Somethign is down"
+#        }
+
+#        self.vip.pubsub.publish('pubsub', topic='platform/send_email',headers=headers ,message=message)
+
+
+    def _test_smtp_address(self, smtp_address,smtp_port,smtp_username,smtp_password):
+        try:
+            server = smtplib.SMTP(self.current_config.get('smtp_address', None),self.current_config.get('smtp_port', None))
+            server.ehlo()
+            server.starttls()
+            #stmplib docs recommend calling ehlo() before & after starttls()
+            server.ehlo()
+            server.login(self.current_config.get('smtp_username', None), self.current_config.get('smtp_password', None))
+            #server.sendmail(SENDER, RECIPIENT, msg.as_string())
+            server.close()
+        except  Exception as e:
+            print ("Error: ", e)
+        else:
+            print ("Email sent!")
+
+    def configure_main(self, config_name, action, contents):
+        """
+        The main configuration callback.
+
+        :param config_name:
+        :param action:
+        :param contents:
+        """
+        self.current_config = self.default_config.copy()
+        self.current_config.update(contents)
+
+        self.current_config['allow_frequency_seconds'] = self.current_config.get(
+            'allow_frequency_minutes', 60) * 60
+
+        smtp_address = self.current_config.get('smtp_address', None)
+        smtp_port = self.current_config.get('smtp_port', None)
+        smtp_username = self.current_config.get('smtp_username', None)
+        smtp_password = self.current_config.get('smtp_password', None)
+
+        if action == "NEW":
```

Comment:
This should be anytime an update happens...as you can change username/password etc.  Once the with gevent passes then you can set them to the member variables for usage.